go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -203,13 +205,20 @@ func (ncc *networkClusterController) syncNodeClusterSubnet(node *corev1.Node) er
 		klog.Warningf("Failed to get node %s host subnets annotations for network %s : %v", node.Name, ncc.networkName, err)
 	}
 
+	networkID, err := util.ParseNetworkIDAnnotation(node, ncc.networkName)
+	if err != nil && !util.IsAnnotationNotSetError(err) {
+		// Log the error and try to allocate new subnets
+		klog.Warningf("Failed to get node %s network id annotations for network %s : %v", node.Name, ncc.networkName, err)
```

Comment:
networkClusterController instance is created for each network.

Cluster manager creates for the default network and it passes a network ID.  For default network its 0.

Secondary network manager will create an instance of this struct for each layer3 secondary network and it generates a unique id for each secondary network and passes this.  So networkID can't be nul.

The warning message you're refering to will be logged if a nodes network id annotation is missing.  In which case it will store the network's id in the network id annotation.

Reply:
> networkClusterController instance is created for each network.
> 
> Cluster manager creates for the default network and it passes a network ID. For default network its 0.
> 
> Secondary network manager will create an instance of this struct for each layer3 secondary network and it generates a unique id for each secondary network and passes this. So networkID can't be nul.

no what I mean is:
```
networkID, err := util.ParseNetworkIDAnnotation(node, ncc.networkName)
	if err != nil && !util.IsAnnotationNotSetError(err) {
```
here err is not nil so something went wrong when trying to parse networkIDAnnotation and you are ensuring the annotation not set error is excluded; even then if err is not nil its possible for networkID to be nil at this point right?
(gah later... I realise you are using the ncc.networkID not the networkID you get from this call to update node annotation although at this point not sure what is in ncc.networkID?)

> 
> The warning message you're refering to will be logged if a nodes network id annotation is missing. In which case it will store the network's id in the network id annotation.

Reply:
ncc.networkID is set when the ncc object/structure is created by secondary cluster manager when a NAD is created.